lib/rollbax/item.ex

```diff
@@ -15,20 +15,75 @@ defmodule Rollbax.Item do
     }
   end
 
-  def compose(draft, {level, msg, time, meta}) do
+  def compose(draft, {level, exception, stacktrace, time, meta}) do
     {occurr_data, meta} =
-      Map.pop(meta, :rollbax_occurr_data, %{})
+      Map.pop(meta, "rollbax_occurr_data", %{})
     Map.update!(draft, "data", fn(data) ->
       Map.merge(occurr_data, data)
-      |> put_body(msg)
+      |> put_body(exception, stacktrace)
       |> put_custom(meta)
       |> Map.put("level", level)
       |> Map.put("timestamp", time)
     end)
   end
 
-  defp put_body(data, msg) do
-    Map.put(data, "body", %{"message" => %{"body" => msg}})
+  # If a message with no stack trace, use "message"
+  defp put_body(data, exception, nil) do
+    Map.put(data, "body", %{ "message" => %{ "body" => Exception.format(:error, exception, nil) } })
+  end
+
+  # If this payload is a single exception, use "trace"
+  # stacktrace example: [{Test, :report, 2, [file: 'file.exs', line: 16]}]
+  defp put_body(data, exception, stacktrace) do
+    Map.put(data, "body", %{ "trace" => trace(exception, stacktrace) })
+  end
+
+  defp trace(exception, stacktrace) do
+    %{ "frames" => frames(stacktrace), "exception" => exception(exception) }
+  end
+
+  # Required: frames
+  # A list of stack frames, ordered such that the most recent call is last in the list.
+  # TODO implement all options. See https://rollbar.com/docs/api/items_post/
+  defp frames(stacktrace) do
+    stacktrace
+    |> Enum.map(fn({_, _, _, f}) -> frame(f[:file], f[:line]) end)
+    |> Enum.filter(&(&1))
```

Comment:
I feel this is hard-ish to read. First of all, I'm not all stacktraces have the form `{_, _, _, _}`. You can likely handle this the same way `Exception.format_stacktrace/0,1` handles this ([relevant code](https://github.com/whatyouhide/elixir/blob/fd37f1f67363be32a65c3ba38f2646ad18266abf/lib/elixir/lib/exception.ex#L339-L374)). Maybe something like this?

```elixir
  defp frames(stacktrace) do
    Enum.map(stacktrace, &stacktrace_entry_to_frame/1)
  end

  def stacktrace_entry_to_frame({module, fun, arity, location}) do
    %{"method" => Exception.format_mfa(module, fun, arity)} |> put_location(location)
  end

  def stacktrace_entry_to_frame({fun, arity, location}) do
    %{"method" => Exception.format_fa(fun, arity)} |> put_location(location)
  end

  defp put_location(frame, location) do
    frame = if file = location[:file], do: Map.put(frame, "filename", file), else: frame
    frame = if line = location[:line], do: Map.put(frame, "lineno", line), else: frame
    frame
  end
```

Wdyt?